src/cascadia/TerminalApp/TabHeaderControl.cpp

```diff
@@ -75,6 +75,14 @@ namespace winrt::TerminalApp::implementation
     void TabHeaderControl::RenameBoxLostFocusHandler(Windows::Foundation::IInspectable const& /*sender*/,
                                                      Windows::UI::Xaml::RoutedEventArgs const& /*e*/)
     {
+        // If the context menu associated with the renamer text box is open we know it gained the focus.
+        // In this case we ignore this event (we will regain the focus once the menu will be closed).
+        const auto flyout = HeaderRenamerTextBox().ContextFlyout();
+        if (flyout && flyout.IsOpen())
```

Comment:
qq: this doesn't make it so if you open the flyout and then navigate away from the box by clicking the terminal or something that the renamer stays open, right?

Reply:
Emm.. this is a default behavior of the  context flyout - to return to the previously focused element when collapsed. I mean, we do not cancel focus loss.

Reply:
Great! Just idiot thoughts

Reply:
Holy crap, I just got autocorrected! I meant to type “idle” thoughts. 😬